Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/Block.cs

```diff
@@ -21,6 +21,19 @@ public static Doc PrintWithConditionalSpace(BlockSyntax node, string groupId)
 
         private static Doc Print(BlockSyntax node, string? groupId)
         {
+            if (
+                node.Statements.Count == 0
```

Comment:
It does work but it seems harder to understand, but maybe that's because of how CSharpier formats it.
```c#
            if (
                node is { Statements: { Count: 0 }, Parent: MethodDeclarationSyntax }
                && !Token.HasComments(node.CloseBraceToken)
            ) {
```
If it was formatted this way it could be easier to understand
```c#
           if (
                node is
                {
                    Statements: { Count: 0 }, 
                    Parent: MethodDeclarationSyntax
                }
                && !Token.HasComments(node.CloseBraceToken)
            ) {
```
It could also be that I'm still somewhat new to reading this type of pattern matching.